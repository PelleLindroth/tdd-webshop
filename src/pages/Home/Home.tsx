import { useEffect, useState } from 'react'
import { IProduct } from '../../mocks/products'
import Card from '../../components/Card/Card'

const Home = (props: { products: IProduct[] }) => {
  const { products } = props
  const [searchPhrase, setSearchPhrase] = useState('')
  const [filteredProducts, setFilteredProducts] = useState<IProduct[]>([])

  useEffect(() => {
    setFilteredProducts(
      products.filter((product) => product.name.includes(searchPhrase))
    )
  }, [products, searchPhrase])

  const handleSearch = (e: React.FormEvent<HTMLInputElement>) => {
    setSearchPhrase(e.currentTarget.value)
  }

  return (
    <main>
      <div>
        <label htmlFor="search">Search product:</label>
        <input
          onChange={handleSearch}
          value={searchPhrase}
          id="search"
          type="search"
        />
      </div>
      {products.length === 0 && <h3>Something went wrong</h3>}
      {filteredProducts.length > 0 ? (
        <ul>
          {filteredProducts.map((product) => (
            <Card product={product} key={product.id} />
          ))}
        </ul>
      ) : (
        <h3>No matching products found</h3>
      )}
    </main>
  )
}

export default Home
