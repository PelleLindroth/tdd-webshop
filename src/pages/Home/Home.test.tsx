import { shallow } from 'enzyme'
import Home from './Home'

describe('Home', () => {
  it('renders Home component correctly', () => {
    const wrapper = shallow(<Home />)

    expect(wrapper).toMatchSnapshot()
  })
})
