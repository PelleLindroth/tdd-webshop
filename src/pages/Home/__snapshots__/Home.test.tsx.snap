// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders Home component correctly 1`] = `
<h2>
  Home
</h2>
`;
